File: Go-Solutions/1304/1304.go
package main

func main() {
	
}

func sumZero(n int) []int {
	if n == 1 {
		return []int{0}
	}
	list := make([]int, 0)
	start := 0
	sum := 0
	for i := 0; i < n - 1; i++ {
		sum += start
		list = append(list, start)
		start++
	}
	list = append(list, -sum)
	result := make([]int, n)
	i := 0
	for _, v := range list {
		result[i] = v
		i++
	}
	return result
}

File: Java-Kotlin/src/main/java/array/_1304.java
package array;

import java.util.ArrayList;
import java.util.List;

/**
 * https://leetcode-cn.com/problems/find-n-unique-integers-sum-up-to-zero/
 */
public class _1304 {

    public int[] sumZero(int n) {
        if (n == 1) {
            return new int[] {0};
        }
        List<Integer> list = new ArrayList<>();
        int start = 0;
        int sum = 0;
        for (int i = 0; i < n - 1; i++) {
            sum += start;
            list.add(start++);
        }
        list.add(-sum);
        int[] result = new int[n];
        int i = 0;
        for (Integer num : list) {
            result[i++] = num;
        }
        return result;
    }
}
